Added environment to create "page" files

julia-zielke opened the pull request:
## Description

- Added a submenu to create a new page under `UI Builder - New Page`
- Added a modal where you only enter the name and a yaml for the page is created 
- Added the new filetype = page
- Added a route for .../page/... as a subpage of explorer 
- Under .../page/... the component PageEditor is loaded

Currently the PageEditor is only a duplicate of the EndpointEditor.

## Checklist

- [ ] Documentation updated if required
- [ ] Test coverage is appropriate
      
## Checklist Internal

- [ ] Linear issue linked (e.g. [DIR-XXXX] pull request title)
- [ ] Has the PR been labeled

julia-zielke commented:
@stefan-kracht When you set the type to "page" in the createFile payload in this [file](src/pages/namespace/Explorer/Tree/components/modals/CreateNew/Page/index.tsx) there is an error, that I couldn't solve yet.

That's the reason why for now I duplicated everything from the EndpointEditor (like the 62 files under /pageEditor/schema and pageEditor/form in that commit [ad18abe]

Maybe you have an idea or we can take a look together tomorrow.

stefan-kracht reviewed (changes requested):
I could not open the file you linked, so I'm not 100% sure what the error was, but let's look at it together later. 

Some other small thing: I know this comes from my suggestion, but when clicking on the new button, should we remove the "UI Builder" menu item and just make it "New Page"? We can divide it further when we start introducing more than one file type (like New Template, etc.). 

But I like the icon you picked for the UI Builder, let's use that for the page! 

We should also update the icon in various places like the breadcrumb, the header of the page builder, and in the file tree. Not sure if we can do all of these already. I think the backend must add `type: page` for the icon to be changeable in the file browser.

julia-zielke commented:
Thanks for the feedback!
Yes, that is what the error was about, that type: page is not valid when submitting the form. So I need the addition in the backend to continue with further customization of the PageEditor.

I changed the menu as you suggested. I agree that we only need a submenu when we have more than one option in it.

julia-zielke commented:
I deleted all the duplicated files from the gateway endpoint and PageEditor on the page URL is for now just an empty component.

I think I can't change the rest like the breadcrumb icon because the data in the payload is type endpoint.
Shall we merge this or wait for the backend to add the filetype=page? What do you think?

stefan-kracht commented on ui/src/pages/namespace/Explorer/Tree/components/modals/CreateNew/Page/index.tsx:
```diff
@@ -0,0 +1,149 @@
+
+  const { mutate: createFile, isPending } = useCreateFile({
+    onSuccess: (data) => {
+      /** To Do:
+       * check if we need to update the Notification Bell for the type = page too?
+       *
+       * creating a new workflow might introduce an uninitialized secret.
+       * We need to update the notification bell, to see potential new messages.
+       */
+      // updateNotificationBell();
```
Should we delete this comment? I think this TODO: will be forgotten if we don't make a ticket, and I think it is not worth making one for now.

stefan-kracht commented on ui/src/assets/locales/en/translation.json:
```diff
@@ -96,6 +96,10 @@
               "label": "Gateway",
               "route": "New Route",
               "consumer": "New Consumer"
+            },
+            "uibuilder": {
+              "label": "UI Builder",
+              "page": "New Page"
```
We have one unused item here. Maybe we make `"New Page"` the `label` and delete the other one?

stefan-kracht commented on ui/src/pages/namespace/Explorer/Page/index.tsx:
```diff
@@ -0,0 +1,53 @@
+
+  if (!namespace) return null;
+  if (!path) return null;
+  if (endpointData?.type !== "endpoint") return null;
```
As I'm reviewing this, there has been an update in the backend from yassir that happened after your last message. I  wiped my local backend to make this changes apply and after that I had to change `endpoint` to `page` to make this page work. 

But with this change, the breadcrumb and the file icon in the file explorer automatically worked. 🤩

stefan-kracht commented on ui/src/api/files/utils.ts:
```diff
@@ -72,6 +80,8 @@ export const fileTypeToIcon = (type: BaseFileSchemaType["type"]) => {
     case "consumer":
       return Users;
+    case "page":
+      return PanelTop;
```
This icon is used for the breadcrumb and the file browser, and it is currently different from the one in the "Create new / New Page" menu. I think it would be nice if this would be the same icon. I leave you the decision on which Icon fits best. I can't decide which one I like better right now :D

stefan-kracht reviewed (changes requested):
I now reviewed everything and made some small suggestions. But after that I think we have a very good starting point

julia-zielke commented:
Thanks for your review! I cleaned everything up and updated the description of the PR and would be ready to merge.
I am also very happy, that Yassir was able to get the backend change in here already. 🙂
